gbn/queue.go

```diff
@@ -197,23 +203,29 @@ func (q *queue) processNACK(seq uint8) (bool, bool) {
 
 	q.cfg.log.Tracef("Received NACK %d", seq)
 
+	var bumped bool
+
 	// If the NACK is the same as sequenceTop, it probably means that queue
 	// was sent successfully, but we just missed the necessary ACKs. So we
 	// can empty the queue here by bumping the base and we dont need to
 	// trigger a resend.
 	if seq == q.sequenceTop {
 		q.sequenceBase = q.sequenceTop
-		return true, false
+		bumped = true
+
+		return false, bumped
```

Comment:
So the reason I changed this, was because it seemed like it was quite confusing the second boolean actually ment in the function prior to this change. It had slipped through previously the function returned `false` here for the "bumped" return boolean, even though it actually did bump. So I just wanted to add this to ensure that there's less chance of that happening again with future changes. I'm totally fine with changing this as you've suggested though if you think that's unnecessary 🙂!

Reply:
yeah I think it is the responsibility of the function comment description to explain what each return value means.

Reply:
Ok, makes sense! Updated :)